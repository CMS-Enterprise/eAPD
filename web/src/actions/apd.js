import { push } from 'connected-react-router';

import { getIsAdmin } from '../reducers/user.selector';
import axios from '../util/api';
import { fromAPI, toAPI } from '../util/serialization/apd';

const LAST_APD_ID_STORAGE_KEY = 'last-apd-id';

export const ADD_APD_KEY_PERSON = 'ADD_APD_KEY_PERSON';
export const CREATE_APD = 'CREATE_APD';
export const CREATE_APD_REQUEST = 'CREATE_APD_REQUEST';
export const CREATE_APD_SUCCESS = 'CREATE_APD_SUCCESS';
export const CREATE_APD_FAILURE = 'CREATE_APD_FAILURE';
export const DELETE_APD_REQUEST = Symbol('delete apd : request');
export const DELETE_APD_SUCCESS = Symbol('delete apd : success');
export const DELETE_APD_FAILURE = Symbol('delete apd : failure');
export const GET_APD_REQUEST = 'GET_APD_REQUEST';
export const GET_APD_SUCCESS = 'GET_APD_SUCCESS';
export const GET_APD_FAILURE = 'GET_APD_FAILURE';
export const REMOVE_APD_KEY_PERSON = 'REMOVE_APD_KEY_PERSON';
export const SAVE_APD_REQUEST = 'SAVE_APD_REQUEST';
export const SAVE_APD_SUCCESS = 'SAVE_APD_SUCCESS';
export const SAVE_APD_FAILURE = 'SAVE_APD_FAILURE';
export const SELECT_APD = 'SELECT_APD';
export const SUBMIT_APD_REQUEST = 'SUBMIT_APD_REQUEST';
export const SUBMIT_APD_SUCCESS = 'SUBMIT_APD_SUCCESS';
export const SUBMIT_APD_FAILURE = 'SUBMIT_APD_FAILURE';
export const UPDATE_APD = 'UPDATE_APD';
export const UPDATE_BUDGET = 'UPDATE_BUDGET';
export const WITHDRAW_APD_REQUEST = Symbol('withdraw apd request');
export const WITHDRAW_APD_SUCCESS = Symbol('withdraw apd success');
export const WITHDRAW_APD_FAILURE = Symbol('withdraw apd failure');

export const SET_SELECT_APD_ON_LOAD = 'SET_SELECT_APD_ON_LOAD';
export const selectApdOnLoad = () => (dispatch, getState) => {
  const isAdmin = getIsAdmin(getState());
  if (!isAdmin) {
    dispatch({ type: SET_SELECT_APD_ON_LOAD });
  }
};

export const addKeyPerson = () => ({ type: ADD_APD_KEY_PERSON });
export const removeKeyPerson = (key, { global = window } = {}) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_KEY_PERSON,
      key
    });
  }
};

export const updateBudget = () => (dispatch, getState) =>
  dispatch({ type: UPDATE_BUDGET, state: getState() });

export const requestApd = () => ({ type: GET_APD_REQUEST });
export const receiveApd = data => ({ type: GET_APD_SUCCESS, data });
export const failApd = error => ({ type: GET_APD_FAILURE, error });

export const updateApd = updates => dispatch => {
  dispatch({ type: UPDATE_APD, updates });
  if (updates.years) {
    dispatch(updateBudget());
  }
};

export const selectApd = (
  id,
  { deserialize = fromAPI, global = window, pushRoute = push } = {}
) => dispatch =>
  axios.get(`/apds/${id}`).then(req => {
    dispatch({ type: SELECT_APD, apd: deserialize(req.data) });
    dispatch(updateBudget());
    dispatch(pushRoute('/apd'));

    if (global.localStorage) {
      global.localStorage.setItem(LAST_APD_ID_STORAGE_KEY, id);
    }
  });

export const createRequest = () => ({ type: CREATE_APD_REQUEST });
export const createSuccess = data => ({ type: CREATE_APD_SUCCESS, data });
export const createFailure = () => ({ type: CREATE_APD_FAILURE });
export const createApd = ({
  deserialize = fromAPI,
  pushRoute = push
} = {}) => dispatch => {
  dispatch(createRequest());
  return axios
    .post('/apds')
    .then(async req => {
      const apd = deserialize(req.data);
      dispatch(createSuccess(apd));
      await dispatch(selectApd(apd.id, { deserialize, pushRoute }));
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch(createFailure(reason));
    });
};

export const requestSave = () => ({ type: SAVE_APD_REQUEST });
export const saveSuccess = data => ({ type: SAVE_APD_SUCCESS, data });
export const saveFailure = data => ({ type: SAVE_APD_FAILURE, data });

export const fetchApd = ({
  global = window,
  pushRoute = push,
  select = selectApd
} = {}) => (dispatch, getState) => {
  dispatch(requestApd());

  const url = `/apds`;

  return axios
    .get(url)
    .then(async req => {
      const apd = Array.isArray(req.data) ? req.data : null;
      dispatch(receiveApd(apd.filter(({ status }) => status !== 'archived')));

      const {
        apd: { selectAPDOnLoad }
      } = getState();
      if (
        selectAPDOnLoad &&
        global.localStorage &&
        global.localStorage.getItem(LAST_APD_ID_STORAGE_KEY)
      ) {
        await dispatch(
          select(global.localStorage.getItem('last-apd-id'), {
            global,
            pushRoute
          })
        );
      }
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch(failApd(reason));
    });
};

const shouldFetchApd = state => !state.apd.loaded;

export const fetchApdDataIfNeeded = () => (dispatch, getState) => {
  if (shouldFetchApd(getState())) {
    return dispatch(fetchApd());
  }

  return null;
};

export const saveApd = ({ serialize = toAPI } = {}) => (dispatch, state) => {
  const {
    apd: { data: updatedApd },
    activities,
    dirty
  } = state();

  if (!dirty.dirty) {
    return Promise.resolve();
  }

  dispatch(requestSave());

  const apd = serialize(updatedApd, activities);

  // These are the fields we want to remove if they aren't dirty.
  const filterAPDFields = [
    'federalCitations',
    'incentivePayments',
    'narrativeHIE',
    'narrativeHIT',
    'narrativeMMIS',
    'programOverview',
    'previousActivityExpenses',
    'previousActivitySummary',
    'stateProfile'
  ];
  const filterActivityFields = [
    'contractorResources',
    'costAllocation',
    'costAllocationNarrative',
    'expenses',
    'goals',
    'schedule',
    'standardsAndConditions',
    'statePersonnel',
    'quarterlyFFP'
  ];

  filterAPDFields.forEach(field => {
    if (!dirty.data.apd[field]) {
      delete apd[field];
    }
  });

  for (let i = apd.activities.length - 1; i >= 0; i -= 1) {
    const activity = apd.activities[i];
    if (!dirty.data.activities.byKey[activity.key]) {
      // If this activity isn't dirty, strip it from the list to save,
      // but then add just its ID back - this prevents it from being
      // deleted by the API synchronization.
      apd.activities.splice(i, 1);
      apd.activities.push({ id: activity.id });
    } else {
      const dirtyActivity = dirty.data.activities.byKey[activity.key];
      // Get rid of any activity fields that aren't dirty.
      filterActivityFields.forEach(field => {
        if (!dirtyActivity[field]) {
          delete activity[field];
        }
      });
    }
  }

  return axios
    .put(`/apds/${updatedApd.id}`, apd)
    .then(res => {
      dispatch(saveSuccess(res.data));
      return res.data;
    })
    .catch(error => {
      if (error.response.status === 403) {
        dispatch(saveFailure('save-apd.not-logged-in'));
      } else {
        dispatch(saveFailure());
      }
      throw error;
    });
};

export const deleteApd = (id, { fetch = fetchApd } = {}) => dispatch => {
  dispatch({ type: DELETE_APD_REQUEST });

  return axios
    .delete(`/apds/${id}`)
    .then(() => {
      dispatch({ type: DELETE_APD_SUCCESS });
      dispatch(fetch());
    })
    .catch(() => {
      dispatch({ type: DELETE_APD_FAILURE });
    });
};
