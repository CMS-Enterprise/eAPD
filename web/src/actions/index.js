export const INCREMENT_COUNT = 'INCREMENT_COUNT';
export const DECREMENT_COUNT = 'DECREMENT_COUNT';

export const increment = () => ({ type: INCREMENT_COUNT });
export const decrement = () => ({ type: DECREMENT_COUNT });
