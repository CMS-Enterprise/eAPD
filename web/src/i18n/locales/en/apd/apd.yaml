title: Program Summary
helpText: >-
  Provide an introduction for the state program. Include context for this funding request by providing information such as current size of program, future vision, and how the program is positioned within the state. This information is similar to the content from the IAPD executive summary template section.
overview:
  title: Program Overview
  helpText: >-
    Provide a brief, high level summary of the state program including history, and future vision. Include history and summary of future vision of the state program.
  reminder: >-

  yearsCovered: Choose the federal fiscal year(s) this APD covers.

introduction:
  title: Program Introduction
  helpText: >-
    Provide a brief, high level summary of the state program, history, and future vision.

hit:
  title: HIT Overview
  helpText: >-
    Summarize briefly planned HIT funded activities contained within this
    APD.
  reminder: >-

hie:
  title: HIE Overview
  helpText: >-
    Summarize planned HIE funded activities within this APD.

    (1) Description of the state's HIE approach (statewide, sub-state
    health information office (HIOs), etc.);

    (2) Anticipated risks and mitigation strategies;

    (3) The short and long term value proposition to providers;

    (4) The role of state government in governance and policy-setting.
  reminder: >-
    CMS recommends states connect to regional CMS HITECH contacts prior to developing HIE funding request. HIE funding requests are complex. Early consultation with CMS facilitates a more efficient approval process.
    (If there are no HIE activities, skip this section.)

mmis:
  title: MMIS Overview
  helpText: >-
    Briefly summarize planned MMIS funded activities contained within this APD.
  reminder: >-
    If there are no MMIS activities, skip this section.
