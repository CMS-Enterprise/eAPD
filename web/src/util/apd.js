import {
  MEDICAID_BUSINESS_AREAS_DISPLAY_LABEL_MAPPING,
  UPDATE_STATUS_LABEL_MAPPING
} from '@cms-eapd/common';

export const updateStatusChoices = updateStatus => {
  let list = [];
  Object.keys(updateStatus).map(key => {
    list.push({
      label: UPDATE_STATUS_LABEL_MAPPING[key],
      value: key,
      checked: updateStatus[key]
    });
  });
  return list;
};

export const businessAreaChoices = medicaidBusinessAreas => {
  let list = [];
  Object.keys(medicaidBusinessAreas).map(key => {
    if (key != 'otherMedicaidBusinessAreas') {
      list.push({
        label: MEDICAID_BUSINESS_AREAS_DISPLAY_LABEL_MAPPING[key],
        value: key,
        checked: medicaidBusinessAreas[key]
      });
    }
  });
  return list;
};

export const arrayOfObjectsToStringList = array => {
  let arrayForList = [];
  // move all choices (checked true) to list
  array.map(obj => {
    if (obj.checked) {
      arrayForList.push(obj.label);
    }
  });

  return arrayForList.join(', ');
};
