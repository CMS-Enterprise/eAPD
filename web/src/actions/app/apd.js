import { push } from 'connected-react-router';

import {
  CREATE_APD_FAILURE,
  CREATE_APD_REQUEST,
  CREATE_APD_SUCCESS,
  DELETE_APD_FAILURE,
  DELETE_APD_REQUEST,
  DELETE_APD_SUCCESS,
  FETCH_ALL_APDS_FAILURE,
  FETCH_ALL_APDS_REQUEST,
  FETCH_ALL_APDS_SUCCESS,
  SAVE_APD_FAILURE,
  SAVE_APD_REQUEST,
  SAVE_APD_SUCCESS,
  SELECT_APD,
  SET_APD_TO_SELECT_ON_LOAD
} from './symbols';
import { updateBudget } from '../budget';
import { EDIT_APD } from '../editApd/symbols';
import { ariaAnnounceApdLoaded, ariaAnnounceApdLoading } from '../aria';

import { selectApdData } from '../../reducers/apd.selectors';
import {
  selectHasChanges,
  selectPatches
} from '../../reducers/patch.selectors';
import { getIsAdmin } from '../../reducers/user.selector';

import axios from '../../util/api';
import initialAssurances from '../../util/regulations';

const LAST_APD_ID_STORAGE_KEY = 'last-apd-id';

export const saveApd = () => (dispatch, getState) => {
  const state = getState();
  const hasChanges = selectHasChanges(state);

  if (!hasChanges) {
    return Promise.resolve();
  }

  dispatch({ type: SAVE_APD_REQUEST });

  const { id: apdID } = selectApdData(state);
  const patches = selectPatches(state);

  return axios
    .patch(`/apds/${apdID}`, patches)
    .then(res => {
      dispatch({ type: SAVE_APD_SUCCESS, data: res.data });
      return res.data;
    })
    .catch(error => {
      if (error.response && error.response.status === 403) {
        dispatch({ type: SAVE_APD_FAILURE, data: 'save-apd.not-logged-in' });
      } else {
        dispatch({ type: SAVE_APD_FAILURE });
      }
      throw error;
    });
};

export const selectApd = (
  id,
  route,
  { global = window, pushRoute = push } = {}
) => dispatch => {
  dispatch(ariaAnnounceApdLoading());

  return axios.get(`/apds/${id}`).then(req => {
    dispatch({ type: SELECT_APD, apd: req.data });

    // By default, APDs get an empty object for federal citations. The canonical list of citations is in frontend
    // code, not backend. So if we get an APD with no federal citations, set its federal citations to the initial
    // values using an EDIT_APD action. That way the initial values get saved back to the API.
    if (Object.keys(req.data.federalCitations).length === 0) {
      dispatch({
        type: EDIT_APD,
        path: '/federalCitations',
        value: initialAssurances
      });
    }

    dispatch(updateBudget());
    dispatch(pushRoute(route));
    dispatch(ariaAnnounceApdLoaded());

    if (global.localStorage) {
      global.localStorage.setItem(LAST_APD_ID_STORAGE_KEY, id);
    }
  });
};

export const setApdToSelectOnLoad = () => (dispatch, getState) => {
  const isAdmin = getIsAdmin(getState());
  if (!isAdmin) {
    dispatch({ type: SET_APD_TO_SELECT_ON_LOAD });
  }
};

export const createApd = ({ pushRoute = push } = {}) => dispatch => {
  dispatch({ type: CREATE_APD_REQUEST });
  return axios
    .post('/apds')
    .then(async req => {
      dispatch({ type: CREATE_APD_SUCCESS, data: req.data });
      await dispatch(selectApd(req.data.id, '/apd', { pushRoute }));
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch({ type: CREATE_APD_FAILURE, data: reason });
    });
};

export const fetchAllApds = ({
  global = window,
  pushRoute = push,
  select = selectApd
} = {}) => (dispatch, getState) => {
  dispatch({ type: FETCH_ALL_APDS_REQUEST });

  const url = `/apds`;

  return axios
    .get(url)
    .then(async req => {
      const apd = Array.isArray(req.data) ? req.data : null;
      dispatch({
        type: FETCH_ALL_APDS_SUCCESS,
        data: apd.filter(({ status }) => status !== 'archived')
      });

      const {
        apd: { selectAPDOnLoad }
      } = getState();
      if (
        selectAPDOnLoad &&
        global.localStorage &&
        global.localStorage.getItem(LAST_APD_ID_STORAGE_KEY)
      ) {
        await dispatch(
          select(global.localStorage.getItem('last-apd-id'), {
            global,
            pushRoute
          })
        );
      }
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch({ type: FETCH_ALL_APDS_FAILURE, error: reason });
    });
};

export const deleteApd = (id, { fetch = fetchAllApds } = {}) => dispatch => {
  dispatch({ type: DELETE_APD_REQUEST });

  return axios
    .delete(`/apds/${id}`)
    .then(() => {
      dispatch({ type: DELETE_APD_SUCCESS });
      dispatch(fetch());
    })
    .catch(() => {
      dispatch({ type: DELETE_APD_FAILURE });
    });
};
