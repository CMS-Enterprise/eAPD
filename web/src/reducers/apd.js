import u from 'updeep';

import {
  ADD_APD_KEY_PERSON,
  GET_APD_REQUEST,
  GET_APD_SUCCESS,
  GET_APD_FAILURE,
  REMOVE_APD_KEY_PERSON,
  UPDATE_APD
} from '../actions/apd';
import { nextSequence } from '../util';

const newPerson = id => ({
  id,
  title: '',
  desc: '',
  years: {
    2018: { amt: '', perc: '' },
    2019: { amt: '', perc: '' }
  }
});

const initialState = {
  data: {
    id: '',
    years: ['2018'],
    overview: '',
    keyPersonnel: [newPerson(1), newPerson(2), newPerson(3)],
    hitNarrative: '',
    hieNarrative: '',
    mmisNarrative: ''
  },
  fetching: false,
  loaded: false,
  error: ''
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_APD_REQUEST:
      return { ...state, fetching: true, error: '' };
    case GET_APD_SUCCESS:
      return {
        ...state,
        fetching: false,
        loaded: true,
        data: {
          id: action.data.id,
          years: ['2018'], // TODO
          overview: action.data.programOverview,
          keyPersonnel: [], // TODO
          hitNarrative: action.data.narrativeHIT,
          hieNarrative: action.data.narrativeHIE,
          mmisNarrative: action.data.narrativeMMIS
        }
      };
    case GET_APD_FAILURE:
      return { ...state, fetching: false, error: action.error };
    case ADD_APD_KEY_PERSON:
      return u(
        {
          data: {
            keyPersonnel: people => [
              ...people,
              newPerson(nextSequence(people.map(p => p.id)))
            ]
          }
        },
        state
      );
    case REMOVE_APD_KEY_PERSON:
      return u(
        {
          data: {
            keyPersonnel: people => people.filter(p => p.id !== action.id)
          }
        },
        state
      );
    case UPDATE_APD:
      return u({ data: { ...action.updates } }, state);
    default:
      return state;
  }
};

export default reducer;
export { newPerson };
