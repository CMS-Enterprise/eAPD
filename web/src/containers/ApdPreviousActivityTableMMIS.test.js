import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import {
  plain as ApdPreviousActivityTableMMIS,
  mapStateToProps,
  mapDispatchToProps
} from './ApdPreviousActivityTableMMIS';
import { updateApd } from '../actions/apd';

describe('apd previous activity table, mmis component', () => {
  const props = {
    previousActivityExpenses: {
      '1': {
        mmis: {
          90: {
            federalActual: 10,
            totalApproved: 20
          },
          75: {
            federalActual: 30,
            totalApproved: 40
          },
          50: {
            federalActual: 50,
            totalApproved: 60
          }
        }
      },
      '2': {
        mmis: {
          90: {
            federalActual: 100,
            totalApproved: 200
          },
          75: {
            federalActual: 300,
            totalApproved: 400
          },
          50: {
            federalActual: 500,
            totalApproved: 600
          }
        }
      }
    },
    updateApd: sinon.spy()
  };

  beforeEach(() => {
    props.updateApd.resetHistory();
  });

  test('renders correctly', () => {
    expect(
      shallow(<ApdPreviousActivityTableMMIS {...props} />)
    ).toMatchSnapshot();
  });

  test('dispatches on a change', () => {
    shallow(<ApdPreviousActivityTableMMIS {...props} />)
      .find('DollarField[name="approved-total-mmis90-1"]')
      .simulate('change', { target: { value: 'new value' } });

    expect(
      props.updateApd.calledWith({
        previousActivityExpenses: {
          '1': { mmis: { '90': { approvedTotal: 'new value' } } }
        }
      })
    );
  });

  test('maps state to props', () => {
    const state = {
      apd: {
        data: {
          previousActivityExpenses: 'previous activities'
        }
      }
    };

    expect(mapStateToProps(state)).toEqual({
      previousActivityExpenses: 'previous activities'
    });
  });

  test('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({ updateApd });
  });
});
