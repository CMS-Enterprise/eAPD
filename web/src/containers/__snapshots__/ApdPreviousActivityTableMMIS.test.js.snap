// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd previous activity table, mmis component renders correctly 1`] = `
<Fragment>
  <h4
    aria-hidden="true"
    className="ds-h4"
  >
    MMIS
  </h4>
  <table
    className="budget-table"
    key="90"
  >
    <caption
      className="ds-h4 ds-u-visibility--screen-reader"
    >
      MMIS 
      <span>
        Federal share 
        <strong>
          90
          %
        </strong>
         FFP
      </span>
    </caption>
    <thead>
      <tr>
        <th
          id="prev_act_mmis_ffy"
        >
          <span
            className="ds-u-visibility--screen-reader"
          >
            Year
          </span>
        </th>
        <th
          id="prev_act_mmis90_total"
        >
          <span>
            Total computable
            <br />
            (Federal + state)
          </span>
        </th>
        <th
          colSpan="2"
          id="prev_act_mmis90_federal"
        >
          <span>
            Federal share 
            <strong>
              90
              %
            </strong>
             FFP
          </span>
        </th>
      </tr>
      <tr>
        <td
          aria-hidden="true"
          className="th"
        />
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis90_total_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis90_federal_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis90_federal_actual"
        >
          Actual
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <th
          id="prev_act_mmis_row_1"
        >
          FFY 1
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis90_total prev_act_mmis90_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 90/10 level for FFY 1, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis90-1"
            onChange={[Function]}
            value={20}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_1 prev_act_mmis90_federal prev_act_mmis90_federal_approved"
        >
          <Dollars
            long={false}
          >
            18
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis90_federal prev_act_mmis90_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 90/10 level for FFY 1"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis90-1"
            onChange={[Function]}
            value={10}
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <th
          id="prev_act_mmis_row_2"
        >
          FFY 2
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis90_total prev_act_mmis90_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 90/10 level for FFY 2, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis90-2"
            onChange={[Function]}
            value={200}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_2 prev_act_mmis90_federal prev_act_mmis90_federal_approved"
        >
          <Dollars
            long={false}
          >
            180
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis90_federal prev_act_mmis90_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 90/10 level for FFY 2"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis90-2"
            onChange={[Function]}
            value={100}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="budget-table"
    key="75"
  >
    <caption
      className="ds-h4 ds-u-visibility--screen-reader"
    >
      MMIS 
      <span>
        Federal share 
        <strong>
          75
          %
        </strong>
         FFP
      </span>
    </caption>
    <thead>
      <tr>
        <th
          id="prev_act_mmis_ffy"
        >
          <span
            className="ds-u-visibility--screen-reader"
          >
            Year
          </span>
        </th>
        <th
          id="prev_act_mmis75_total"
        >
          <span>
            Total computable
            <br />
            (Federal + state)
          </span>
        </th>
        <th
          colSpan="2"
          id="prev_act_mmis75_federal"
        >
          <span>
            Federal share 
            <strong>
              75
              %
            </strong>
             FFP
          </span>
        </th>
      </tr>
      <tr>
        <td
          aria-hidden="true"
          className="th"
        />
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis75_total_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis75_federal_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis75_federal_actual"
        >
          Actual
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <th
          id="prev_act_mmis_row_1"
        >
          FFY 1
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis75_total prev_act_mmis75_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 75/25 level for FFY 1, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis75-1"
            onChange={[Function]}
            value={40}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_1 prev_act_mmis75_federal prev_act_mmis75_federal_approved"
        >
          <Dollars
            long={false}
          >
            30
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis75_federal prev_act_mmis75_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 75/25 level for FFY 1"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis75-1"
            onChange={[Function]}
            value={30}
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <th
          id="prev_act_mmis_row_2"
        >
          FFY 2
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis75_total prev_act_mmis75_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 75/25 level for FFY 2, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis75-2"
            onChange={[Function]}
            value={400}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_2 prev_act_mmis75_federal prev_act_mmis75_federal_approved"
        >
          <Dollars
            long={false}
          >
            300
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis75_federal prev_act_mmis75_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 75/25 level for FFY 2"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis75-2"
            onChange={[Function]}
            value={300}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="budget-table"
    key="50"
  >
    <caption
      className="ds-h4 ds-u-visibility--screen-reader"
    >
      MMIS 
      <span>
        Federal share 
        <strong>
          50
          %
        </strong>
         FFP
      </span>
    </caption>
    <thead>
      <tr>
        <th
          id="prev_act_mmis_ffy"
        >
          <span
            className="ds-u-visibility--screen-reader"
          >
            Year
          </span>
        </th>
        <th
          id="prev_act_mmis50_total"
        >
          <span>
            Total computable
            <br />
            (Federal + state)
          </span>
        </th>
        <th
          colSpan="2"
          id="prev_act_mmis50_federal"
        >
          <span>
            Federal share 
            <strong>
              50
              %
            </strong>
             FFP
          </span>
        </th>
      </tr>
      <tr>
        <td
          aria-hidden="true"
          className="th"
        />
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis50_total_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis50_federal_approved"
        >
          Approved
        </th>
        <th
          className="ds-u-text-align--right"
          id="prev_act_mmis50_federal_actual"
        >
          Actual
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <th
          id="prev_act_mmis_row_1"
        >
          FFY 1
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis50_total prev_act_mmis50_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 50/50 level for FFY 1, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis50-1"
            onChange={[Function]}
            value={60}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_1 prev_act_mmis50_federal prev_act_mmis50_federal_approved"
        >
          <Dollars
            long={false}
          >
            30
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_1 prev_act_mmis50_federal prev_act_mmis50_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 50/50 level for FFY 1"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis50-1"
            onChange={[Function]}
            value={50}
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <th
          id="prev_act_mmis_row_2"
        >
          FFY 2
        </th>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis50_total prev_act_mmis50_total_approved"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="total approved funding for MMIS at the 50/50 level for FFY 2, state plus federal"
            labelClassName="ds-u-visibility--screen-reader"
            name="approved-total-mmis50-2"
            onChange={[Function]}
            value={600}
          />
        </td>
        <td
          className="budget-table--number"
          headers="prev_act_mmis_row_2 prev_act_mmis50_federal prev_act_mmis50_federal_approved"
        >
          <Dollars
            long={false}
          >
            300
          </Dollars>
        </td>
        <td
          className=""
          headers="prev_act_mmis_row_2 prev_act_mmis50_federal prev_act_mmis50_federal_actual"
        >
          <DollarField
            className="budget-table--input-holder"
            fieldClassName="budget-table--input__number"
            label="actual federal share for MMIS at the 50/50 level for FFY 2"
            labelClassName="ds-u-visibility--screen-reader"
            name="actual-federal-mmis50-2"
            onChange={[Function]}
            value={500}
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
